src/generic/trilinos_eigen_solver.h

```diff
@@ -765,6 +854,116 @@ namespace oomph
       }
     }
 
+    /// Solve the adjoint eigen problem
+    void solve_adjoint_eigenproblem(Problem* const& problem_pt,
+                                    const int& n_eval,
+                                    Vector<std::complex<double>>& eigenvalue,
+                                    Vector<DoubleVector>& eigenvector)
+    {
+      // No access to sigma, so set from sigma real
+      Sigma = Sigma_real;
+
+      // Initially be dumb here
+      Linear_solver_pt = problem_pt->linear_solver_pt();
+
+      // Let's make the initial vector
+      Teuchos::RCP<DoubleMultiVector> initial = Teuchos::rcp(
+        new DoubleMultiVector(1, problem_pt->dof_distribution_pt()));
+      Anasazi::MultiVecTraits<double, DoubleMultiVector>::MvRandom(*initial);
+
+      // Make the operator
+      // NB Using AdjointProblemBasedShiftInvertOperator, the only difference
+      // to solve_eigenproblem
+      Teuchos::RCP<DoubleMultiVectorOperator> Op_pt =
+        Teuchos::rcp(new AdjointProblemBasedShiftInvertOperator(
+          problem_pt, this->linear_solver_pt(), Sigma));
+
+      // Create the basic problem
+      Teuchos::RCP<Anasazi::BasicEigenproblem<double,
+                                              DoubleMultiVector,
+                                              DoubleMultiVectorOperator>>
+        anasazi_pt = Teuchos::rcp(
+          new Anasazi::BasicEigenproblem<double,
+                                         DoubleMultiVector,
+                                         DoubleMultiVectorOperator>(Op_pt,
+                                                                    initial));
+
+      // Think I have it?
```

Comment:
Not sure what the comment is implying, i.e. what is "it"? (I'd guess from the usage that it means that you already have the Hermitian, and it doesn't need to be set, but it's not entirely clear.)

Reply:
Can't remember what the comment in meant to imply, but the command is simply to indicate that the problem is not Hermitian in general, so change comment to reflect that.

Reply:
Changed comment to something useful